tools/flakeguard/README.md

```diff
@@ -0,0 +1,60 @@
+# Flakeguard
+
+**Flakeguard** is a tool designed to help identify flaky tests within a Go project. Flaky tests are tests that intermittently fail without changes to the code, often due to race conditions or other non-deterministic behavior. Flakeguard assists by analyzing the impact of code changes on test packages and by running tests multiple times to determine stability.
+
+## Features
+
+- **Identify Impacted Tests**: Detects test packages that may be affected by changes in your Go project files.
+- **Run Tests for Flakiness**: Runs tests multiple times to determine their flakiness.
+- **Output Results in JSON**: Allows easy integration with CI pipelines and custom reporting.
+- **Supports Exclusion Lists**: Configurable to exclude specified packages or paths from the analysis.
+- **Recursive Dependency Analysis**: Detects all impacted packages through dependency levels.
+
+## Installation
+
+To install `flakeguard` CLI, you need to have Go installed on your machine. With Go installed, run the following command:
+
+```sh
+go install github.com/smartcontractkit/chainlink-testing-framework/tools/flakeguard@latest
+```
+
+## Usage
+
+Flakeguard offers two main commands: `find` and `run`.
+
+### `find` Command
+
+The `find` command identifies test packages affected by recent changes.
+
+#### Flags for `find` Command
```

Comment:
These are redundant if this is just copying your help command, and risks going out of date. I generally prefer if you suggest people run with the `--help` flag to see this. Or if you want, you could do a `go generate` to help update this for you?

Reply:
@kalverra yeah, that make sense. Let me remove it and suggest --help flag

Reply:
fixed